### PR TITLE
Make cheats available in hints

### PR DESCRIPTION
**Pull request by kasperski95**
Based on https://github.com/software-mansion/protostar/pull/68

- injected all cheats to `hint_locals`

### CODE REVIEW

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
LGTM some questions in comments

File: src/commands/test/cheatcodes/cheats.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+from typing import TYPE_CHECKING, Any, Dict, List
+
+from starkware.starknet.public.abi import get_selector_from_name
+from typing_extensions import TypedDict
+
+if TYPE_CHECKING:
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply src/commands/test/cheatcodes/cheats.py:6]: Importing it normally causes circular dependency error.

File: src/commands/test/cheatcodes/cheats.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    from src.commands.test.cheatcodes.syscall_handler import CheatableSysCallHandler
+    from src.commands.test.runner_test import TestRunner
+else:
+    TestRunner = Any
+    CheatableSysCallHandler = Any
+
+
+class ModifiedUnits(TypedDict):
+    test_runner: TestRunner
+    cheatable_syscall_handler: CheatableSysCallHandler
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply src/commands/test/cheatcodes/cheats.py:16]: `TypedDict` it's first what came to my mind. I'll make it use `dataclass`.

File: src/commands/test/cheatcodes/cheats.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kasperski95) [reply src/commands/test/cheatcodes/cheats.py]: Code is now refactored. Resolving.

**Comment by kasperski95** [comment]
@MaksymilianDemitraszek Can you re-review this PR?

File: src/commands/test/runner_with_cheatcodes.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    )
+    async def run_tests_in(self, *args, **kwargs):
+        original_run_from_entrypoint = CairoFunctionRunner.run_from_entrypoint
+        CairoFunctionRunner.run_from_entrypoint = (
+            self._get_run_from_entrypoint_with_custom_hint_locals(
+                CairoFunctionRunner.run_from_entrypoint
+            )
+        )
+        await super().run_tests_in(*args, **kwargs)
+        CairoFunctionRunner.run_from_entrypoint = original_run_from_entrypoint
```
[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply src/commands/test/runner_with_cheatcodes.py:24]: Probably we have to do the same for `@replace_class` can you please include it here?

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
LGTM, close to being merge, one minor thing to fix

File: src/commands/test/runner_with_cheatcodes.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kasperski95) [reply src/commands/test/runner_with_cheatcodes.py:24]: Ok, I'll do it

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Pull request merged**